aspnetcore/mvc/models/file-uploads.md

```diff
@@ -1,195 +1,654 @@
+* **Run a virus/malware scanner on uploaded content before the file is stored.**
+
+&dagger;The sample app demonstrates an approach that meets the criteria.
+
+> [!WARNING]
+> Uploading malicious code to a system is frequently the first step to executing code that can:
+>
+> * Completely takeover a system.
+> * Overload a system with the result that the system crashes.
+> * Compromise user or system data.
+> * Apply graffiti to a public UI.
+>
+> For information on reducing the attack surface area when accepting files from users, see the following resources:
+>
+> * [Unrestricted File Upload](https://www.owasp.org/index.php/Unrestricted_File_Upload)
+> * [Azure Security: Ensure appropriate controls are in place when accepting files from users](/azure/security/azure-security-threat-modeling-tool-input-validation#controls-users)
+
+For more information on implementing security measures, including examples from the sample app, see the [Validation](#validation) section.
+
+## Storage scenarios
+
+Common storage options for files include:
+ 
+* Database
+
+  * For small file uploads, often faster than physical storage (file system or network share) options.
+  * Often more convenient than physical storage options because retrieval of a database record for user data can concurrently supply the file content (for example, an avatar image).
+  * Potentially less expensive than using a data storage service.
+
+* Physical storage (file system or network share)
+
+  * For large file uploads:
+    * Database limits may restrict the size of the upload.
+    * Often less economical than storage in a database.
+  * Potentially less expensive than using a data storage service.
+  * The app's process must have read and write permissions to the storage location. **Never grant execute permission.**
+
+* Data storage service (for example, [Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/))
+
+  * Improved scalability and resiliency over on-premises solutions that are often subject to single points of failure.
+  * Potentially lower cost in large storage infrastructure scenarios.
+
+  For more information, see [Quickstart: Use .NET to create a blob in object storage](/azure/storage/blobs/storage-quickstart-blobs-dotnet). The topic demonstrates <xref:Microsoft.Azure.Storage.File.CloudFile.UploadFromFileAsync*>, but you can use <xref:Microsoft.Azure.Storage.File.CloudFile.UploadFromStreamAsync*> to save a <xref:System.IO.FileStream> to blob storage when working with a <xref:System.IO.Stream>.
+
+## File upload scenarios
+
+Two general approaches for uploading files are buffering and streaming.
+
+**Buffering**
+
+The entire file is read into an <xref:Microsoft.AspNetCore.Http.IFormFile>, which is a C# representation of the file used to process or save the file.
+
+Any single buffered file exceeding 64 KB is moved from memory to a temp file on disk.
```

Comment:
Management generally doesn't like them and has ordered that they mostly not be used. However, I think you're correct here for that reason ... and another important reason: It can lead to deadly resource issues on the server and the app doesn't report anything ... the server just blows up in your face. I'll put it back into a NOTE.